src/Auxiliary/Aux_ComputeProfile.cpp

```diff
@@ -120,28 +128,42 @@ void Aux_ComputeProfile( Profile_t *Prof[], const double Center[], const double
    if ( PatchType != PATCH_LEAF  &&  PatchType != PATCH_NONLEAF  &&
         PatchType != PATCH_BOTH  &&  PatchType != PATCH_LEAF_PLUS_MAXNONLEAF )
       Aux_Error( ERROR_INFO, "incorrect PatchType (%d) !!\n", PatchType );
-#  endif
 
+#  ifdef OPENMP
+   if (  omp_in_parallel()  &&  ( (PrepTimeIn >= 0.0 && !OPT__INT_TIME) ||
```

Comment:
This check prevents users from applying temporal interpolation after computing the profiles (e.g., the GREP profiles in the `ccsn` branch). How about printing a warning message instead? Or solve the thread-unsafe issue (`OPT__INT_TIME`)?

Reply:
@hfhsieh I'm confused. In the `ccsn` branch, where we do launch multiple threads to _call_ this function in parallel? I thought we only apply OpenMP _inside_ this function.